django/views/defaults.py

```diff
@@ -40,6 +43,12 @@ def page_not_found(request, exception, template_name='404.html'):
         body = template.render(context, request)
         content_type = None             # Django will use DEFAULT_CONTENT_TYPE
     except TemplateDoesNotExist:
+        if template_name != '404.html':
```

Comment:
I suggest you use a constant to refer to the default `template_name`. Have a look at how it was dealt with in #5577.